test/archive.cpp

```diff
@@ -280,6 +280,123 @@ TEST(ZimArchive, openSplitZimArchive)
+    ASSERT_EQ(ref_item.getClusterIndex(), test_item.getClusterIndex());               \
+    ASSERT_EQ(ref_item.getBlobIndex(), test_item.getBlobIndex());                     \
+    ASSERT_EQ(ref_item.getData(), test_item.getData());                               \
+  }
+
+TEST(ZimArchive, cacheDontImpactReading)
+{
+  const TestCacheConfig cache_configs[] = {
+    {0, 0, 0},
+    {1, 1, 1},
+    {2, 2, 2},
+    {10, 10, 10},
+    {1000, 2000, 1000},
+    {0, 2000, 1000},
+    {1000, 0, 1000},
+    {1000, 2000, 0},
+    {1, 2000, 1000},
+    {1000, 1, 1000},
+    {1000, 2000, 1},
+  };
+
+  for (auto& testfile: getDataFilePath("small.zim")) {
+    auto ref_archive = zim::Archive(testfile.path);
+
+    for (auto cache_config: cache_configs) {
+      auto test_archive = zim::Archive(testfile.path);
+      test_archive.set_dirent_cache_max_size(cache_config.dirent_cache_size);
+      test_archive.set_dirent_lookup_cache_max_size(cache_config.dirent_lookup_cache_size);
+      test_archive.set_cluster_cache_max_size(cache_config.cluster_cache_size);
+
+      EXPECT_EQ(test_archive.get_dirent_cache_max_size(), cache_config.dirent_cache_size);
+      EXPECT_EQ(test_archive.get_dirent_lookup_cache_max_size(), cache_config.dirent_lookup_cache_size);
+      EXPECT_EQ(test_archive.get_cluster_cache_max_size(), cache_config.cluster_cache_size);
+
+      ASSERT_ARCHIVE_EQUIVALENT(ref_archive, test_archive)
+    }
+  }
+}
+
+
+TEST(ZimArchive, cacheChange)
+{
+  for (auto& testfile: getDataFilePath("wikibooks_be_all_nopic_2017-02.zim")) {
+    std::cout << testfile.path << std::endl;
+    auto ref_archive = zim::Archive(testfile.path);
+    auto archive = zim::Archive(testfile.path);
+
+    archive.set_dirent_cache_max_size(30);
+    archive.set_cluster_cache_max_size(5);
+
+    auto range = ref_archive.iterEfficient();
+    auto ref_it = range.begin();
+    ASSERT_ARCHIVE_EQUIVALENT_IT_LIMIT(ref_it, range.end(), archive, 50)
+    EXPECT_EQ(archive.get_dirent_cache_current_size(), 30);
+    EXPECT_EQ(archive.get_cluster_cache_current_size(), 2); // Only 2 clusters in the file
+
+    // Reduce cache size
+    archive.set_dirent_cache_max_size(10);
+    archive.set_cluster_cache_max_size(1);
+
+    EXPECT_EQ(archive.get_dirent_cache_current_size(), 10);
+    EXPECT_EQ(archive.get_cluster_cache_current_size(), 1);
+
+    ASSERT_ARCHIVE_EQUIVALENT_IT_LIMIT(ref_it, range.end(), archive, 50)
+
+    EXPECT_EQ(archive.get_dirent_cache_current_size(), 10);
+    EXPECT_EQ(archive.get_cluster_cache_current_size(), 1);
+
+    // Clean cache
+    // (More than testing the value, this is needed as we want to be sure the cache is actually populated later)
+    archive.set_dirent_cache_max_size(0);
+    archive.set_cluster_cache_max_size(0);
+
+    EXPECT_EQ(archive.get_dirent_cache_current_size(), 0);
+    EXPECT_EQ(archive.get_cluster_cache_current_size(), 0);
+
+    // Increase the cache
```

Comment:
With an `ASSERT_ARCHIVE_EQUIVALENT()` easy/concise check at your service, doesn't it make sense to check that thing still work after decreasing the cache size to 0?

Reply:
The purpose of this test is to test than changing the cache size has an effect. Not that it doesn't change the returned value. `ASSERT_ARCHIVE_EQUIVALENT` is "just" a nice to have to be sure, but not the main purpose of the test.

Reply:
> The purpose of this test is to test than changing the cache size has an effect.

That's prone to misinterpretation and I also wanted to ask if the commit's description was wrong. Yes, we want to be sure that setting the cache size actually changes the cache size, but I think that we also want to test that changing the cache size on a live archive doesn't break its operation (i.e. in that sense, it has no observable effect other than on performance).